### PR TITLE
Fixes #27969 - Use GNU tar to list files

### PR DESCRIPTION
**Pull request by mbacovsky**
Rubygems TarReader has some problems with huge archives.
Switching to GNU tar.

### CODE REVIEW

**Comment by johnpmitsch** [comment]
It looks fine to me! What do you think about adding a test to validate the regex returns what we expect?

I likely won't have time to test functionality until Friday because of the meetup in Raleigh, so feel free to have someone else test and approve :)

**Comment by mbacovsky** [comment]
@johnpmitsch updated. I finally realized we don't need any regexp as the non-verbose list (tar -tf <arch>) prints just the filenames. Test added anyway. Also removed dependency on a Feature, we probably don't want this in lib where we do not load definitions while testing.

@upadhyeammit, @kgaikwad, could you please take a look?

File: lib/foreman_maintain/utils/backup.rb [hunk 1]
```diff
@@ -213,16 +213,7 @@ def incremental?
...
-        file_list = []
-        File.open(tarball, 'rb') do |file|
-          ::Zlib::GzipReader.wrap(file) do |gz|
-            ::Gem::Package::TarReader.new(gz) do |tar|
-              tar.each { |entry| file_list << entry.full_name }
-            end
-          end
-        end
-        file_list
+        execute("tar -tf #{tarball}").split("\n")
```
Comment (kgaikwad) [inline lib/foreman_maintain/utils/backup.rb]: I think this method can be used when there are multiple config files to verify.
But this method is used only by `validate_hostname?` method to verify config file `etc/httpd/conf/httpd.conf` file exists or not. 

So why not to use `tar` command directly?
For example -  `execute?("tar -tf #{tarball}  #{file_name}")`

What do you think?

Comment (mbacovsky) [reply lib/foreman_maintain/utils/backup.rb]: I like the idea

**Comment by mbacovsky** [comment]
@kgaikwad I reduced the the tar calls and removed the list method. Is that acceptable?

**Comment by jameerpathan111** [comment]
> @jameerpathan111, could you please test it once to confirm?

Sure, will test it soon.

**Comment by jameerpathan111** [comment]
@kgaikwad 

The PR has been successfully verified on actual environment(Customer DB upgrade setup).
Test steps:
1- Reproduced the issue mentioned in [BZ1754881](https://bugzilla.redhat.com/show_bug.cgi?id=1754881)
2- Used this PR to see if issue is fixed or not.

Observation:
- foreman-maintain restore completed successfully.

**Comment by mbacovsky** [comment]
Thanks @kgaikwad and @jameerpathan111!

**Pull request merged**